src/backend/core/tests/documents/test_api_documents_list.py

```diff
@@ -532,3 +534,40 @@ def test_api_documents_list_filter_link_reach_invalid():
         ]
     }
 
+# Filters: title
+
+@pytest.mark.parametrize("query,nb_results", [
+    ("Project Alpha", 1),       # Exact match
+    ("project", 2),             # Partial match (case-insensitive)
+    ("Guide", 1),               # Word match within a title
+    ("Special", 0),             # No match (nonexistent keyword)
+    ("2024", 2),                # Match by numeric keyword
+    ("", 5),                    # Empty string
+])
+def test_api_documents_list_filter_link_reach(query, nb_results):
+    """Authenticated users should be able to search documents by their title."""
```

Comment:
Will you be able to find any public or authenticated document as well? I guess it should not.

Reply:
As I understand it, the search is restricted to the current "list" view, so you will not have public/restricted results here. 

Anyway, it could be explicitly tested here, just to be sure ^^

Reply:
you are filtering only among the results you can see if this is the question. Is it the question?

Reply:
I added this test to secure that filtering and permissions don't interfere: https://github.com/numerique-gouv/impress/pull/429/commits/1e1fce5fa75e1760f619d76d34794f23e12411af